addons/website_sale/models/website.py

```diff
@@ -772,12 +772,19 @@ def _get_and_cache_current_cart(self):
         SaleOrderSudo = self.env['sale.order'].sudo()
 
         sale_order_sudo = SaleOrderSudo
-        partner_sudo = self.env.user.partner_id
         if CART_SESSION_CACHE_KEY in request.session:
             sale_order_sudo = SaleOrderSudo.browse(request.session[CART_SESSION_CACHE_KEY])
+
+            try:
+                # fetch the record field or raise a missingError
+                # avoids a query with the use of exists
+                sale_order_sudo and sale_order_sudo.state
+            except MissingError:
+                self.sale_reset()
+                sale_order_sudo = SaleOrderSudo
```

Comment:
Honestly, I believe this kind of hack to avoid one query is too much.
It makes the code way less readable and standard and I doubt it's worth it :thinking:.